### PR TITLE
Adding System.IO.Compression.ZipFile to net471 support package

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @safern @AlexGhiondea 

These changes will add System.IO.Compression.ZipFile to the support package and will cause a conflict so that the right binding redirect is generated for it.

### CODE REVIEW

File: src/System.IO.Compression.ZipFile/ref/System.IO.Compression.ZipFile.csproj [hunk 1]
```diff
@@ -8,10 +8,19 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net471-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net471-Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netfx-Release|AnyCPU'" />
   <ItemGroup>
-    <Compile Include="System.IO.Compression.ZipFile.cs" />
+    <Compile Include="System.IO.Compression.ZipFile.cs" Condition="'$(TargetGroup)' != 'net471' And '$(TargetGroup)' != 'netfx'" />
```
Comment (weshaggard) [inline src/System.IO.Compression.ZipFile/ref/System.IO.Compression.ZipFile.csproj]: Might be a little easier to understand if you just move these compile items under the ItemGroups with the matching references.

File: src/System.IO.Compression.ZipFile/ref/Configurations.props [hunk 2]
```diff
@@ -4,6 +4,8 @@
     <BuildConfigurations>
       netcoreapp;
       uap;
+      netfx;
```
Comment (weshaggard) [inline src/System.IO.Compression.ZipFile/ref/Configurations.props]: Just double check that this doesn't change what is included for net461.

File: src/System.IO.Compression.ZipFile/ref/System.IO.Compression.ZipFile.csproj (hunk 1 above)
Comment (joperezr) [reply src/System.IO.Compression.ZipFile/ref/System.IO.Compression.ZipFile.csproj]: sounds good. I'll change it.

File: src/System.IO.Compression.ZipFile/ref/Configurations.props (hunk 2 above)
Comment (joperezr) [reply src/System.IO.Compression.ZipFile/ref/Configurations.props]: It turns out we were overriding the net461 with this change, so with my latest change that is now fixed.

**Comment by joperezr** [comment]
One more change that has to go in, this looks like it removed ZipFile from 4.6.1, I'll fix that and send one more commit

**Comment by joperezr** [comment]
Ok with my final change the resulting package seems to be correct now.

File: pkg/NETStandard.Library.NETFramework/NETStandard.Library.NETFramework.pkgproj [hunk 3]
```diff
@@ -42,7 +42,11 @@
 
   <Target Name="IncludeNETStandardShims">
     <ItemGroup>
-      <_libFileNames Include="@(LibFile->'%(FileName)')" Condition="'%(Extension)' == '.dll'" />
+      <!-- 
+          System.IO.Compression is a special case because we do want to include the netstandard shim for net461, but in net471 we want to
```
Comment (weshaggard) [inline pkg/NETStandard.Library.NETFramework/NETStandard.Library.NETFramework.pkgproj]: System.IO.Compression -> System.IO.Compression.ZipFile

**Pull request merged**